apps/aecore/lib/aecore/peers/sync.ex

```diff
@@ -189,15 +189,14 @@ defmodule Aecore.Peers.Sync do
     case Chain.has_block?(block_hash) do
       false ->
         case(HttpClient.get_block({peer_uri, block_hash})) do
-          {:ok, peer_block} ->
-            deserialized_block = Serialization.block(peer_block, :deserialize)
+          {:ok, deserialized_block} ->
             try do
               BlockValidation.single_validate_block(deserialized_block)
               peer_block_hash =
                 BlockValidation.block_header_hash(deserialized_block.header)
 
               if(block_hash == Base.encode16(peer_block_hash)) do
-                check_peer_block(peer_uri, peer_block.header.prev_hash,
+                check_peer_block(peer_uri, Serialization.hex_binary(deserialized_block.header.prev_hash, :serialize),
```

Comment:
Here I had to do a little bit dirty fix, because we use hex encoded hashes in requests and hex encoded hash is passed to `check_peer_block`

Reply:
internally I'd prefer to only pass binary, we should only introduce other encodings at the last point needed